internal/provider/resource_fmc_access_control_policy.go

```diff
@@ -156,28 +380,65 @@ func (r *AccessControlPolicyResource) Create(ctx context.Context, req resource.C
 
 	// Create object
 	body := plan.toBody(ctx, AccessControlPolicy{})
+	bodyCats := gjson.Parse(body).Get("dummy_categories").Array()
+	body, _ = sjson.Delete(body, "dummy_categories")
+	bodyRules := gjson.Parse(body).Get("dummy_rules").Array()
+	body, _ = sjson.Delete(body, "dummy_rules")
 	res, err := r.client.Post(plan.getPath(), body, reqMods...)
 	if err != nil {
 		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to configure object (POST), got error: %s, %s", err, res.String()))
 		return
 	}
 	plan.Id = types.StringValue(res.Get("id").String())
-	res, err = r.client.Get(plan.getPath()+"/"+url.QueryEscape(plan.Id.ValueString()), reqMods...)
+
+	err = r.createCatsAt(ctx, plan, bodyCats, 0, reqMods...)
 	if err != nil {
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to retrieve object (GET), got error: %s, %s", err, res.String()))
+		resp.Diagnostics.AddError("Client Error", err.Error())
+
+		res, err = r.client.Delete(plan.getPath()+"/"+url.QueryEscape(plan.Id.ValueString()), reqMods...)
+		if err != nil {
+			resp.Diagnostics.AddWarning("Client Error", fmt.Sprintf("Also, cannot DELETE a hanging policy object, got error: %s, %s", err, res.String()))
+		}
+
 		return
 	}
-	plan.updateFromBody(ctx, res)
 
-	tflog.Debug(ctx, fmt.Sprintf("%s: Create finished successfully", plan.Id.ValueString()))
+	err = r.createRulesAt(ctx, plan, bodyRules, 0, reqMods...)
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", err.Error())
+
+		res, err = r.client.Delete(plan.getPath()+"/"+url.QueryEscape(plan.Id.ValueString()), reqMods...)
+		if err != nil {
+			resp.Diagnostics.AddWarning("Client Error", fmt.Sprintf("Also, cannot DELETE a hanging policy object, got error: %s, %s", err, res.String()))
+		}
+
+		return
+	}
 
 	diags = resp.State.Set(ctx, &plan)
 	resp.Diagnostics.Append(diags...)
-}
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("%s: Create finished successfully, starting implicit Read", plan.Id.ValueString()))
+
+	readReq := resource.ReadRequest{
+		State:        resp.State,
+		ProviderMeta: req.ProviderMeta,
+	}
+	readResp := resource.ReadResponse{
+		State:   resp.State,
+		Private: resp.Private,
+	}
+	r.Read(ctx, readReq, &readResp)
```

Comment:
Just to confirm, what do we need the implicit read for? To populate the UUIDs?

Reply:
Yes, we need UUID of every category and every rule. The DELETE works only with UUIDs (DELETE has a `filter` and as far as I've researched it only works with UUIDs; if it worked with names, all the code would be much simpler).

Reply:
I am just thinking if it is worth getting those UUIDs from the POST responses instead of doing a full read?

Reply:
Done, thanks.